torch/onnx/symbolic_helper.py

```diff
@@ -192,6 +193,46 @@ def wrapper(g, *args, **kwargs):
         return wrapper
     return decorator
 
+def quantized_args(*arg_q_descriptors, op_scale=None, op_zero_point=None):
+    def decorator(fn):
+        fn._op_scale = op_scale
+        fn._op_zero_point = op_zero_point
+
+        @wraps(fn)
+        def wrapper(g, *args, **kwargs):
+            _op_scale = fn._op_scale
+            if _op_scale is not None:
+                _op_scale = g.op("Constant", value_t=torch.tensor(_op_scale))
+            _op_zero_point = fn._op_zero_point
+            if _op_zero_point is not None:
+                _op_zero_point = g.op("Constant", value_t=torch.tensor(_op_zero_point))
+
+            # some args may be optional, so the length may be smaller
+            assert len(arg_q_descriptors) >= len(args)
+            get_desc_args = lambda : zip(arg_q_descriptors[:len(args)], args)
+            # Run regular symbolic function if none of the argument is QTensor.
+            if not any([(desc and arg.node().kind() == "prim::TupleConstruct") for desc, arg in get_desc_args()]):
+                return fn(g, *args, **kwargs)
+
+            dequantized_args = []
+            for desc, arg in get_desc_args():
+                if desc:
+                    dequantized_arg, scale, zero_point = _dequantize_helper(g, arg)
+                    dequantized_args.append(dequantized_arg)
+                    if _op_scale is None:
+                        _op_scale = scale
+                    if _op_zero_point is None:
+                        _op_zero_point = zero_point
+                else:
+                    dequantized_args.append(arg)
+            # TODO: only support single output
+            output = fn(g, *dequantized_args, **kwargs)
+
+            assert _op_scale is not None
```

Comment:
Seems a bit paranoid to be asserting this when the code that assures this is not None is just a few lines above. I personally would remove it.

Reply:
if None it will cause segfault, as `_quantize_helper` calls `.type()` on `scale`. Moving the assertion to `_quantize_helper`.